crates/sui-graphql-rpc/src/types/object.rs

```diff
@@ -1167,8 +1218,14 @@ impl Loader<HistoricalKey> for Db {
                 continue;
             }
 
-            let object =
-                Object::try_from_stored_history_object(stored.clone(), key.checkpoint_viewed_at)?;
+            let object = Object::try_from_stored_history_object(
+                stored.clone(),
+                key.checkpoint_viewed_at,
+                // This conversion will use the object's own version as the `Object::root_version`
+                // if it's not a child object, but if it is, it will be set to `None`, since we do
+                // not have any information on the `root_version` of the fetched dynamic object.
```

Comment:
nit: this comment needs to updated as well now, right? Because we'll always set to the object's own version if this is `None`.

Reply:
Good catch